Extractor/Types/TypeConverter.cs

```diff
@@ -0,0 +1,344 @@
+using System;
```

Comment:
I know it's a refactor, so naturally the feeling is that the total coverage should not change (-0.09% is negligent), but I have a feeling that the new `TypeConverter.cs` could be covered a bit more to raise the confidence.
Currently it shows 70%, perhaps raising it to 80%?

Reply:
I may be able to get it up a little, by handling some of the partials, but if you look at the coverage in codecov there's a fair bit of exception handling. This is purely defensive programming, and shouldn't be possible to accomplish, but the SDK where we get these types is so opaque and complicated that I can't just assume that it all goes well.

I doubt I'll be able to get it up above 80%, but I can try covering a few more built-in types.

Reply:
I did get it above 80 by adding a few more test cases.